apps/mull-ui/src/app/components/event-card/event-card.tsx

```diff
@@ -30,6 +31,7 @@ export const EventCard = ({ event, style = {} }: EventCardProps) => {
         // TODO: Replace placeholder
         src="https://www.citywindsor.ca/residents/parksandforestry/City-Parks/PublishingImages/Assumption%20Park%20Street%20View.JPG"
         alt="Event"
+        onClick={onClick}
```

Comment:
I did some experimentation and found a successful way to handle nested onclicks. just add a call to `e.stopPropagation` to the child `onClick`. This will allow you to set `onClick` on the event-card-container div without shadowing the join button.

Here's my suggestion:
```tsx
// 1. Remove the onclick you added

// 2. modify the container div as such:
<div className="event-card-container" style={style} onClick={onClick}>
...

// 3. Modify the join button onClick as such:
<button
  onClick={(e) => {
    e.stopPropagation();
    setJoined(!joined);
  }}
  className={`event-card-join ${joined ? 'joined' : ''}`}
>
```